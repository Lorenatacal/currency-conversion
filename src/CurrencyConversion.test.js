import React from 'react';
import ReactDOM from 'react-dom';
import axios from 'axios';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import CurrencyConvertor from './CurrencyConversion';
import waait from 'waait';

Enzyme.configure ({adapter: new Adapter ()});
jest.mock ('axios');

describe ('CurrencyConvertor()', () => {
  test ('should render correctly without data', () => {
    const wrapper = Enzyme.shallow (<CurrencyConvertor />);
    expect (toJson (wrapper)).toMatchSnapshot ();
  });

  test ('should render correctly with data', async () => {
    const mockedResponse = {
      data: {
        USD: 3.81,
        EUR: 3.36,
      },
    };
    axios.get.mockResolvedValue (mockedResponse);

    const handleSubmitSpy = jest.spyOn (
      CurrencyConvertor.prototype,
      'handleSubmit'
    );
    const wrapper = Enzyme.shallow (<CurrencyConvertor />);

    const form = wrapper.find ('[data-name="submit"]');
    form.simulate ('submit', {
      preventDefault: () => {},
    });

    expect (handleSubmitSpy).toHaveBeenCalled ();
    expect (axios.get).toHaveBeenCalledWith ('/front-end-test-api', {
      headers: {
        accept: 'application/json',
      },
      params: {
        amount: '',
      },
    });
    await waait ();
    expect (toJson (wrapper)).toMatchSnapshot ();
  });
});
