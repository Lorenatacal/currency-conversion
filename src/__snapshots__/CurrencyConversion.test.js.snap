// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyConvertor() should render correctly with data 1`] = `
<div>
  <form
    data-name="submit"
    onSubmit={[Function]}
  >
    <span>
      Enter GBP amount: 
    </span>
    <input
      max="10000"
      min="1"
      onChange={[Function]}
      type="number"
    />
    <input
      type="submit"
      value="Submit"
    />
  </form>
  <ul>
    <li
      key="0"
    >
      <span>
        USD
        :
      </span>
      <span>
         
        3.81
      </span>
    </li>
    <li
      key="1"
    >
      <span>
        EUR
        :
      </span>
      <span>
         
        3.36
      </span>
    </li>
  </ul>
  <button
    onClick={[Function]}
  >
    Sort A-Z
  </button>
</div>
`;

exports[`CurrencyConvertor() should render correctly without data 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <span>
      Enter GBP amount: 
    </span>
    <input
      max="10000"
      min="1"
      onChange={[Function]}
      type="number"
    />
    <input
      data-name="submit"
      type="submit"
      value="Submit"
    />
  </form>
  <ul />
  <button
    onClick={[Function]}
  >
    Sort A-Z
  </button>
</div>
`;
